src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final int vaultOfClasses = 5;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomValue = random.nextInt(vaultOfClasses);
+        int index = random.nextInt(5);
+        if (index == 0) {
+            return new Circle(colorSupplier.getRandomColor(), randomValue);
+        }
+        if (index == 1) {
+            return new Square(colorSupplier.getRandomColor(), randomValue);
+        }
+        if (index == 2) {
+            return new Rectangle(colorSupplier.getRandomColor(), randomValue, randomValue);
+        }
+        if (index == 3) {
+            return new RightTriangle(colorSupplier.getRandomColor(), randomValue, randomValue);
+        }
+        return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                randomValue, randomValue, randomValue);
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle("White", 10);
```

Comment:
Also think about magic numbers and constants. [Common mistake](https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#all-magic-numbers-in-your-code-should-be-constants).